### PR TITLE
SC-2747: fixes a bug in receipt handling

### PR DESCRIPTION
**Pull request by godsflaw**
This commit fixes a bug in the receipt handling for a transaction that is never executed on-chain.  The receipt comes back as `None` in this case, which means we cannot test `success` on it.  This commit fixes that.

### CODE REVIEW

File: tests/test_keeper_chief.py [hunk 1]
```diff
@@ -108,3 +114,50 @@ def test_check_hat(self, mcd: DssDeployment, keeper: ChiefKeeper, guy_address: A
         assert self.spell.eta() != 0
 
 
+    def test_check_eta_receipt(
+        self, mcd: DssDeployment,
+        keeper: ChiefKeeper,
+        simpledb: SimpleDatabase,
+        our_address: Address
+    ):
```
Comment (KentonPrescott) [inline tests/test_keeper_chief.py:122]: For consistency with the other tests, please keep the test method interface to a single line.

File: tests/test_keeper_chief.py [hunk 2]
```diff
@@ -108,3 +114,50 @@ def test_check_hat(self, mcd: DssDeployment, keeper: ChiefKeeper, guy_address: A
...
+
+        hat = mcd.ds_chief.get_hat()
+
+        etas = keeper.database.db.get(doc_id=3)['upcoming_etas']
+        verify([hat.address], etas, 1)
+
+        keeper.check_eta()
+
+        # Confirm that the spell was casted and that the database was updated
+        assert DSSBadSpell(mcd.web3, Address(hat)).done() == False
```
Comment (KentonPrescott) [inline tests/test_keeper_chief.py:161]: Could you add a little more context either in this comment or as a comment near the test method header that points out the modification in `cast()`? 

Someone with familiarity of how `casts` are supposed to function in a properly written `DSSSpell`  may be quick to assume that this comment, and thus test, must be a false positive.

File: tests/test_keeper_chief.py (hunk 1 above)
Comment (godsflaw) [reply tests/test_keeper_chief.py:122]: or, change them all to fit within 80 characters. 😏

File: tests/test_keeper_chief.py (hunk 2 above)
Comment (godsflaw) [reply tests/test_keeper_chief.py:161]: good point, I'll add more context.

**Comment by godsflaw** [comment]
```
.                                              [100%]

---------- coverage: platform darwin, python 3.7.4-final-0 -----------
Name                           Stmts   Miss  Cover
--------------------------------------------------
chief_keeper/__init__.py           0      0   100%
chief_keeper/chief_keeper.py     113     11    90%
chief_keeper/database.py          70      0   100%
chief_keeper/spell.py             27      2    93%
--------------------------------------------------
TOTAL                            210     13    94%


============================ 11 passed in 20.16 seconds ============================
```

File: tests/test_keeper_chief.py (hunk 1 above)
Comment (godsflaw) [reply tests/test_keeper_chief.py]: style revert

File: tests/test_keeper_chief.py (hunk 2 above)
Comment (godsflaw) [reply tests/test_keeper_chief.py:156]: added more context

**Pull request merged**